tests/phpunit/elementor/modules/global-classes/test-global-classes-usage.php

```diff
@@ -0,0 +1,260 @@
+							'breakpoint' => 'desktop',
+							'state' => null,
+						],
+						'props' => [
+							'color' => [
+								'$$type' => 'color',
+								'value' => 'blue',
+							]
+						],
+					],
+				],
+			],
+		],
+		'order' => ['g-4-124', 'g-4-123'],
+	];
+
+	private $mock_elementor_data = [
+		[
+			'id' => 'abc123',
+			'elType' => 'e-div-block',
+			'settings' => [
+				'classes' => [
+					'value' => ['g-4-123', 'e-4-124']
+				]
+			],
+			'elements' => [
+				[
+					'id' => 'def456',
+					'elType' => 'e-div-block',
+					'settings' => [
+						'classes' => [
+							'value' => ['g-4-124', 'g-4-123', 'e-4-1222']
+						]
+					],
+					'elements' => [
+						[
+							'id' => 'ghi789',
+							'elType' => 'e-heading',
+							'settings' => [
+								'classes' => [
+									'value' => ['g-4-123', 'e-4-1222']
+								]
+							],
+						],
+						[
+							'id' => 'jkl10122',
+							'elType' => 'widget',
+							'widgetType' => 'e-heading',
+							'settings' => [],
+						],
+						[
+							'id' => 'jkl101',
+							'elType' => 'widget',
+							'widgetType' => 'heading',
+							'settings' => [],
+						]
+					],
+				],
+			],
+		],
+	];
+
+	private $mock_elementor_data_2 = [
+		[
+			'id' => 'abc1234',
+			'elType' => 'e-div-block',
+			'settings' => [
+				'classes' => [
+					'value' => ['g-4-123', 'e-4-124']
+				]
+			],
+			'elements' => [
+				[
+					'id' => 'def4564',
+					'elType' => 'e-div-block',
+					'settings' => [
+						'classes' => [
+							'value' => ['g-4-124', 'e-4-1222']
+						]
+					],
+					'elements' => [
+						[
+							'id' => 'ghi7894',
+							'elType' => 'e-button',
+							'settings' => [
+								'classes' => [
+									'value' => ['g-4-123', 'e-4-1222']
+								]
+							],
+						],
+						[
+							'id' => 'jkl1014',
+							'elType' => 'widget',
+							'widgetType' => 'heading',
+							'settings' => [],
+						]
+					],
+				],
+			],
+		],
+	];
+
+	public function setUp(): void {
+		parent::setUp();
+
+		Plugin::$instance->kits_manager->create_new_kit( 'kit' );
+
+		Global_Classes_Repository::make()->put(
+			$this->mock_global_classes['items'],
+			$this->mock_global_classes['order'],
+		);
+
+		remove_all_filters( 'elementor/tracker/send_tracking_data_params' );
+	}
+
+	public function tear_down() {
+		( new Global_Classes_Repository() )->put( [], [] );
+
+
+		foreach ( $this->post_ids as $post_id ) {
+			wp_delete_post( $post_id, true );
+		}
+
+		$this->post_ids = [];
+
+		parent::tear_down();
+	}
+
+	public function test_global_classes_usage() {
+		// Arrange.
+		$global_classes_usage = new Global_Classes_Usage();
+		$global_classes_usage->register_hooks();
+		$this->make_mock_post_with_elements( $this->mock_elementor_data );
+
+		// Act.
+		$params = [];
+
+		$params = apply_filters( 'elementor/tracker/send_tracking_data_params', $params );
+
+		// Assert.
+		$this->assertEquals( 2, $params['usages']['global_classes']['total_count'] );
+		$this->assertEquals(
```

Comment:
```suggestion
		$this->assertSame( 2, $params['usages']['global_classes']['total_count'] );
		$this->assertSame(
```

Just to be stricter
also others